### PR TITLE
Update CEL documentation to capture type checking errors.

### PR DESCRIPTION
**Pull request by whitneygriffith**
Part of [#47533](https://github.com/istio/istio/issues/47533)

### CODE REVIEW

**Review by zirain — APPROVED** [review]

File: envoy/extensions/stackdriver/config/v1alpha1/config.proto [hunk 1]
```diff
@@ -127,7 +127,14 @@ message PluginConfig {
 
   // CEL expression for filtering access logging. If the expression evaluates
   // to true, an access log entry will be generated. Otherwise, no access log
-  // entry will be generated.
+  // entry will be generated. If there are any type errors, the CEL expression is evaluated as false.  
+	// More details on type checking can be found https://kubernetes.io/docs/reference/using-api/cel/#type-checking.
```
Comment (hzxuzhonghu) [inline envoy/extensions/stackdriver/config/v1alpha1/config.proto:131]: indents

Comment (whitneygriffith) [reply envoy/extensions/stackdriver/config/v1alpha1/config.proto]: Fixed!

File: envoy/extensions/stackdriver/config/v1alpha1/config.proto [hunk 2]
```diff
@@ -127,7 +127,14 @@ message PluginConfig {
...
   // to true, an access log entry will be generated. Otherwise, no access log
-  // entry will be generated.
+  // entry will be generated. If there are any type errors, the CEL expression is evaluated as false.
+  // More details on type checking can be found https://kubernetes.io/docs/reference/using-api/cel/#type-checking.
+  // A common error is referring to a non-existent field in the log entry.
+  // It's crucial to note that in Envoy, the fields that appear in access log entries can vary. This variation is influenced by several factors,
+  // including the protocol in use (such as HTTP or TCP), the applied filters, and the specific configuration of the Envoy instance.
+  // Therefore, when using CEL expressions for filtering access logs, it's essential to ensure that these expressions accurately refer to existing fields in the log entry.
+  // The has() macro in CEL may be used in CEL expressions to check if a field is accessible before attempting to access the field's value. 
+  // There's also the CEL Playground https://playcel.undistro.io/ where CEL expressions can be quickly tested.
```
Comment (craigbox) [inline envoy/extensions/stackdriver/config/v1alpha1/config.proto:137]:
```suggestion
  // entry will be generated. If there are any type errors, the CEL expression
  // is evaluated as false. More details on type checking can be found 
  // at https://kubernetes.io/docs/reference/using-api/cel/#type-checking.
  // A common error is referring to a non-existent field in the log entry.
  // It's crucial to note that in Envoy, the fields that appear in access log
  // entries can vary. This variation is influenced by several factors,
  // including the protocol in use (such as HTTP or TCP), the applied filters,
  // and the specific configuration of the Envoy instance. Therefore, when 
  // using CEL expressions for filtering access logs, it's essential to ensure
  // that the expressions accurately refer to existing fields in the log entry.
  // The has() macro in CEL may be used in CEL expressions to check if a field 
  // is accessible before attempting to access the field's value. 
  // You can also quickly test CEL expressions at the CEL Playground 
  // at https://playcel.undistro.io/.
```

Comment (craigbox) [reply envoy/extensions/stackdriver/config/v1alpha1/config.proto:137]: slight rewording/typos, plus reformatting to 80 cols per [protobuf style guide](https://protobuf.dev/programming-guides/style/)

**Review by craigbox — APPROVED** [review]
LGTM. @istio/technical-oversight-committee for approval

**Review by howardjohn — APPROVED** [review]

**Pull request merged**